src/Layer/OrientedImageLayer.js

```diff
@@ -0,0 +1,166 @@
+import * as THREE from 'three';
+import GeometryLayer from 'Layer/GeometryLayer';
+import OrientedImageMaterial from 'Renderer/OrientedImageMaterial';
+import ImmersiveControls from 'Controls/ImmersiveControls';
+import GeoJsonParser from 'Parser/GeoJsonParser';
+import CameraCalibrationParser from 'Parser/CameraCalibrationParser';
+import Coordinates from 'Core/Geographic/Coordinates';
+import OrientationUtils from 'Utils/OrientationUtils';
+
+const coord = new Coordinates('EPSG:4978', 0, 0, 0);
+
+function updatePano(context, camera, layer) {
+    // look for the closest oriented image
+    let minD = Infinity;
+    let minI = -1;
+    for (let i = 0; i < layer.panos.length; i++) {
+        const position = layer.panos[i].position;
+        const D = camera.position.distanceTo(position);
+        if (D < minD) {
+            minD = D;
+            minI = i;
+        }
+    }
+    // detection of oriented image change
+    const ori = layer.panos[minI];
+    if (ori && layer.currentPano != ori) {
+        layer.currentPano = ori;
+        ori.index = minI;
+
+        if (context.view.controls && context.view.controls instanceof ImmersiveControls) {
```

Comment:
Could you use frame requester to update the control?

Reply:
No, because I don't want the control to be updated at each frame.

Reply:
It's better to decouple the `controls` and the `layers` (as it is the case in the others). A layer must only manage its data and update it.

Reply:
ok, I add a callback to the layer, so it doesn't have to know the controls